Robot_Adapter/Create/Panels.cs

```diff
@@ -118,7 +80,64 @@ private bool CreateCollection(IEnumerable<Panel> panels)
 
         /***************************************************/
 
-    }
+        private RobotGeoObject CreateRobotContour(List<Edge> edges, out List<Edge> subEdges)
+        {
+            RobotGeoObject contourPanel = null;
+            subEdges = new List<Edge>();
+            foreach (Edge edge in edges) //Explode Edges into sub edges (as BHoM edges can contain polycurves, polylines etc.)
+            {
+                foreach (ICurve crv in BHEG.Query.ISubParts(edge.Curve).ToList())
+                {
+                    subEdges.Add(BH.Engine.Structure.Create.Edge(crv, edge.Support, edge.Release, edge.Name));
+                }
+            }
+            if (subEdges.Count > 1)
+            {
+                contourPanel = m_RobotApplication.CmpntFactory.Create(IRobotComponentType.I_CT_GEO_CONTOUR);
+                foreach (Edge edge in subEdges)
+                {
+                    RobotGeoSegment segment = m_RobotApplication.CmpntFactory.Create(BH.Engine.Robot.Convert.SegmentType(edge.Curve));
+                    (contourPanel as RobotGeoContour).Add(BH.Engine.Robot.Convert.Segment(edge.Curve, segment));
+                }
+            }
+            else
+            {
+                contourPanel = m_RobotApplication.CmpntFactory.Create(IRobotComponentType.I_CT_GEO_CIRCLE);
+                BH.Engine.Robot.Convert.SingleContourGeometry(subEdges[0].Curve, contourPanel);
+            }
+            contourPanel.Initialize();
+            return contourPanel;
+        }
 
-}
+        /***************************************************/
+
+        private bool HasEdgeSupports(List<Edge> edges)
+        {
+            bool hasSupport = false;
+            foreach (Edge edge in edges)
+            {
+                if (edge.Support != null)
+                {
+                    if (edge.Support.Name != "") hasSupport = true;
```

Comment:
Micro optimisation, that shouldn't have any real noticeable impact, but you could return true here rather that storing a bool.

Reply:
Thanks! updated, tested on my machine (all ok) and pushed. Worth a quick run through the test on your machine for completeness.